fabtests/multinode/src/pattern.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2017-2019 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <pattern.h>
+#include <core.h>
+
+static int broadcast_next(int *cur)
+{
+	int next;
+	if (pm_job.my_rank == 0) {
+		next = *cur + 1;
+
+		if (next >= pm_job.num_ranks)
+			return -FI_ENODATA;
+		if (next == 0)
+			next = 1;
+		
+		*cur = next;
+	}
+	else{
+		return -FI_ENODATA;
+	}
+	return 0;
+}
+
+static int broadcast_current(int *cur)
+{
+	int next;
+	if (pm_job.my_rank == 0) {
+		return -FI_ENODATA;
+	} else {
+		next = *cur + 1;
+
+		if (next > 0)
+			return -FI_ENODATA;
+
+		*cur = next;
+	}
+	return 0; 
+}
+
+static int gather_next(int *cur)
+{
+	int next;
+	if (pm_job.my_rank == 0) {
+		next = *cur + 1;
+
+		if (next >= pm_job.num_ranks)
+			return -FI_ENODATA;
+		if (next == 0)
+			next = 1;
+
+		*cur = next;
+	} else {
+		return -FI_ENODATA;
+	}
+	return 0;
+}
+
+static int gather_current(int *cur)
+{
+	int next;
+	if (pm_job.my_rank == 0) {
+		return -FI_ENODATA;
+	} else {
+		next = *cur + 1;
+
+		if (next > 0)			
+			return -FI_ENODATA;
+
+		*cur = next;
+	}
+	return 0; 
+}
+
+static int ring_next(int *cur)
+{
+	if ((pm_job.my_rank == 0 ? 
+		pm_job.num_ranks - 1 : pm_job.my_rank - 1) == *cur) {
```

Comment:
It's confusing to have what is essentially an if statement embedded inside another if statement check.  Please try to re-work.  Also, else statement below should be on same line as closing brace